scripts/benchmarks/convert_pytest_JSON_to_XUBM.py

```diff
@@ -0,0 +1,104 @@
+"""
+This module includes functionality to convert the provided pytest-benchmark JSON file to
+JSON-XUBM (XanadU BenchMarks) format.
+"""
+import argparse, json, os, sys
+########################################################################
+# Parsing arguments
+########################################################################
+def parse_args():
+    """Parse external arguments provided to the script."""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--filename",
+        type=str,
+        default="benchmarks.json",
+        nargs="?",
+        help="Name of the Pytest-benchmark JSON file to be imported.",
+    )
+
+    parser.add_argument(
+        "--filename_XUBM",
+        type=str,
+        default="benchmarks_xubm.json",
+        nargs="?",
+        help="Name of the XUBM JSON file to be exported.",
+    )
+
+    parser.add_argument(
+        "--author",
+        type=str,
+        nargs="?",
+        help="User associated with the commit.",
+    )
+
+    parser.add_argument(
+        "--github_reference",
+        type=str,
+        default="",
+        nargs="?",
+        help="The fully-formed ref of the branch or tag that triggered the workflow run.",
+    )
+
+    return parser.parse_args()
+
+def create_benchmark_XUBM(stored_data, data, args):
+    """This function converts the JSON file provided by pytest-benchmark to the JSON-XUBM (XanadU BenchMark) format.
+
+    Args:
+        stored_data: data we already have in-place or an empty dictionary.
+        data: Pytest-benchmark JSON data
+        args: customization arguments
+
+    Returns:
+        JSON-XUBM: JSON data
+    """
+    commit_reference = args.github_reference if (args.github_reference != "") else data["commit_info"]["branch"]
+    if commit_reference not in stored_data:
+        stored_data[commit_reference] = {}
+    else:
+        print("data already exist. doing nothing.")
+        return stored_data
+
+    hash_commit = hash(data["commit_info"]["time"]) + hash(commit_reference)
+    for benchmark in data["benchmarks"]:
+        benchmark_xubm = {}
+        benchmark_xubm["uid"] = hash_commit + hash(benchmark["fullname"]) + hash(json.dumps(benchmark["params"]))
+        benchmark_xubm["date"] = data["commit_info"]["time"]
+        benchmark_xubm["fullname"] = benchmark["fullname"]
+        benchmark_xubm["gitID"] = data["commit_info"]["id"]
+        benchmark_xubm["runs"] = benchmark["stats"]["iterations"]
+        benchmark_xubm["params"] = benchmark["params"]
+        benchmark_xubm["runtime"] = benchmark["stats"]["mean"]
+        benchmark_xubm["timeUnit"] = "seconds"
+        benchmark_xubm["project"] = data["commit_info"]["project"]
+        benchmark_xubm["machine"] = data["machine_info"]
+        benchmark_xubm["users"] = args.author
```

Comment:
not sure what the XUBM format is but the name "users" made me expect a list instead of a single author, just making sure this doesn't break anything.

Reply:
Nothing to break here. The field ``users`` is general so in the future, if needed, we can have more than one user.
For now, we are using it for a single user.